vpp-patches/0013-session-fix-populating-chained-buffers.patch

```diff
@@ -0,0 +1,42 @@
+From bbee13b6f04515ee42f066f3f0bfc6b675b5a9a3 Mon Sep 17 00:00:00 2001
+From: Ivan Shvedunov <ivan4th@gmail.com>
+Date: Mon, 15 Mar 2021 19:05:14 +0300
+Subject: [PATCH] session: fix populating chained buffers
+
+Type: fix
+
+Signed-off-by: Ivan Shvedunov <ivan4th@gmail.com>
+Change-Id: I3fdc1711d508bc5b0c2f56b3bd3fb4a56056eb54
+---
+ src/vnet/session/session_node.c | 8 +++++---
+ 1 file changed, 5 insertions(+), 3 deletions(-)
+
+diff --git a/src/vnet/session/session_node.c b/src/vnet/session/session_node.c
+index 2fde85b03..0add12b97 100644
+--- a/src/vnet/session/session_node.c
++++ b/src/vnet/session/session_node.c
+@@ -630,16 +630,18 @@ session_tx_fifo_chain_tail (vlib_main_t * vm, session_tx_context_t * ctx,
+ 	      svm_fifo_t *f = ctx->s->tx_fifo;
+ 	      session_dgram_hdr_t *hdr = &ctx->hdr;
+ 	      u16 deq_now;
++	      u32 offset;
++
+ 	      deq_now = clib_min (hdr->data_length - hdr->data_offset,
+ 				  len_to_deq);
+-	      n_bytes_read = svm_fifo_peek (f, hdr->data_offset, deq_now,
+-					    data);
++	      offset = hdr->data_offset + SESSION_CONN_HDR_LEN;
++	      n_bytes_read = svm_fifo_peek (f, offset, deq_now, data);
```

Comment:
Upstream really needs to judge this. I guess it is good enough for us for the moment.

Anyhow, wouldn't this be enough and a much smaller change?
```
	      n_bytes_read = svm_fifo_peek (f, hdr->data_offset + SESSION_CONN_HDR_LEN,
					    deq_now, data);
```

Reply:
I copied the construct from the other part of the code which does similar thing and made it look the same b/c if you have to do some repetition in the code (and in case of VPP that's rather often the case), it better should look as similar as possible. I could have changed the other place too instead, but maybe they did it just to make the lines shorter, so indent doesn't mangle them too much, and compiler should eliminate the difference